docs(self-hosted): provide guide for scaling up taskbroker and taskworker (#15136)
getsentry/sentry-docs: develop-docs/self-hosted/tasks.mdx

Workers use client-side loadbalancing to distribute load across the brokers they
have been assigned to.

If you only want to scale `taskworker` replicas and you have a single `taskbroker`,
you can easily increase it (up to 24 replicas) by modifying your `docker-compose.override.yml`
file:

```yaml
services:
  taskworker:
    deploy:
      replicas: 5
```

This will spawn 5 replicas of the `taskworker` service. Bear in mind that although
the processing capacity is increased, you will need to monitor and scale your
system resources (CPU and RAM) accordingly.

## Scaling brokers

If you have reached the maximum number of `taskworker` replicas, you can scale

> your `taskbroker` replicas. First, you need to increase the number of partitions
> on "taskworker" topic in Kafka, the number of partitions should be greater than
> the number of `taskbroker` replicas that you're planning to scale to.

Reviewer: ```suggestion
your `taskbroker` replicas. First, you need to increase the number of partitions
on "taskworker" topic in Kafka, the number of partitions should be evenly divisible by
the number of `taskbroker` replicas that you're planning to scale to.
```

Then due to the limited capability of Docker Compose, you will need to manually
scale your `taskbroker` replicas. You can do this by adding more container
declarations to your `docker-compose.override.yml` file:

```yaml
services:
  taskbroker-beta:
    restart: "unless-stopped"
    image: "$TASKBROKER_IMAGE"
    environment:
      TASKBROKER_KAFKA_CLUSTER: "kafka:9092"
      TASKBROKER_KAFKA_DEADLETTER_CLUSTER: "kafka:9092"
      TASKBROKER_DB_PATH: "/opt/sqlite/taskbroker-activations-beta.sqlite"
    volumes:
      - sentry-taskbroker:/opt/sqlite
    depends_on:
      - kafka
  taskbroker-charlie:
    restart: "unless-stopped"
    image: "$TASKBROKER_IMAGE"
    environment:
      TASKBROKER_KAFKA_CLUSTER: "kafka:9092"
      TASKBROKER_KAFKA_DEADLETTER_CLUSTER: "kafka:9092"
      TASKBROKER_DB_PATH: "/opt/sqlite/taskbroker-activations-charlie.sqlite"
    volumes:
      - sentry-taskbroker:/opt/sqlite
    depends_on:
      - kafka
```

> Note that each `taskbroker` need their own SQLite database per replica, to prevent
> issues with corrupted SQLite databases.

Reviewer: ```suggestion
Note that each `taskbroker` replica needs their own SQLite database per replica, to prevent
issues contention and locks between replicas.
```
